testsuite/Rakefile

```
@@ -129,4 +146,111 @@ namespace :utils do
+        next
+      else
+        Rake::Task['jenkins:generate_rake_run_step'].invoke("",
+                                                            File.basename(minion,".feature"),
+                                                            "features/build_validation/init_clients",
+                                                            "run_sets/build_validation_init_client_#{minion}.yml")
+        Rake::Task['jenkins:generate_rake_run_step'].reenable
+      end
+    end
+  end
+
+  desc 'Generate rake files for Maintenance Update Channels and repositories'
+  task :generate_rake_files_add_maintenance_update_repository, [:instances] do |_t, args|
+    minions = args[:instances]
+    minions.each do |minion|
+      if minion.include?('ssh')
+        next
+      else
+        Rake::Task['jenkins:generate_rake_run_step'].invoke("_add_maintenance_update_repositories",
+                                                            File.basename(minion,".feature"),
+                                                            "features/build_validation/add_custom_repositories",
+                                                            "run_sets/build_validation_add_maintenance_update_repositories_#{minion}.yml")
+        Rake::Task['jenkins:generate_rake_run_step'].reenable
+      end
+    end
+  end
+
+  desc 'Generate rake files for activation Keys feature'
+  task :generate_rake_files_add_activation_keys , [:instances] do |_t, args|
+    minions = args[:instances]
+    minions.each do |minion|
+      Rake::Task['jenkins:generate_rake_run_step'].invoke("_add_activation_key",
+                                                          File.basename(minion,".feature"),
+                                                          "features/build_validation/add_activation_keys",
+                                                          "run_sets/build_validation_add_activation_key_#{minion}.yml")
+      Rake::Task['jenkins:generate_rake_run_step'].reenable
+    end
+  end
+
+  desc 'Generate rake file for smoke tests '
+  task :generate_rake_files_smoke_tests , [:instances] do |_t, args|
+    minions = args[:instances]
+    minions.each do |minion|
+      if minion.include?('proxy')
+        next
+      else
+        puts "#{minion}"
+        Rake::Task['jenkins:generate_rake_run_step'].invoke("_smoke_tests",
+                                                            File.basename(minion,".feature"),
+                                                            "features/build_validation/smoke_tests",
+                                                            "run_sets/build_validation_smoke_tests_#{minion}.yml")
+        Rake::Task['jenkins:generate_rake_run_step'].reenable
+      end
+    end
+  end
+
+  desc 'Generate all the rake files needed for Build Validation'
+  task :generate_rake_files_build_validation do
+    instances = []
+    ENV.select { |k, v| k.include?("MINION") || k.include?("PROXY") }.each do |key, value|
```

```suggestion
    ENV.select { |k, _v| k.include?("MINION") || k.include?("PROXY") }.each do |key, _value|
```
Both are not used if I am not mistaken.